documentation/start-documenting.rst

```diff
@@ -64,79 +64,109 @@ Building the documentation
 
 .. highlight:: bash
 
-The toolset used to build the docs is written in Python and is called Sphinx_.
-Sphinx is maintained separately and is not included in this tree.  Also needed
-are blurb_, a tool to create :file:`Misc/NEWS` on demand; and
-python-docs-theme_, the Sphinx theme for the Python documentation.
+To build the documentation, follow the steps in one of the sections below.
+You can view the documentation after building the HTML
+by opening the file :file:`Doc/build/html/index.html` in a web browser.
 
-To build the documentation, follow the instructions from one of the sections
-below.  You can view the documentation after building the HTML by pointing
-a browser at the file :file:`Doc/build/html/index.html`.
+.. note::
 
-You are expected to have installed the latest stable version of
-Sphinx_ and blurb_ on your system or in a virtualenv_ (which can be
-created using ``make venv``), so that the Makefile can find the
-``sphinx-build`` command.  You can also specify the location of
-``sphinx-build`` with the ``SPHINXBUILD`` :command:`make` variable.
+   The following instructions all assume your current working dir is
+   the ``Doc`` subdirectory in your :ref:`CPython repository clone <checkout>`.
+   Make sure to switch to it with ``cd Doc`` if necessary.
 
 
-.. _building-using-make:
+.. _doc-create-venv:
+
+Create a virtual environment
+----------------------------
 
-Using make / make.bat
----------------------
+.. _doc-create-venv-unix:
 
-**On Unix**, run the following from the root of your :ref:`repository clone
-<checkout>` to build the output as HTML::
+**On Unix platforms** that support :program:`make`
+(including Linux, macOS and BSD),
+you can create a new :mod:`venv` with the required dependencies using::
 
-   cd Doc
    make venv
```

Comment:
I was having a hard time with `venv` on my ubuntu 16.04 dev machine where I installed and used python3.7 instead of the system default python 3.5.

It seems a quite common issue that people are facing.

https://askubuntu.com/questions/1268833/error-command-path-to-env-bin-python3-7-im-ensurepip-upgrade
```
$ make venv
Error: Command '['/root/repo/cpython/Doc/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
/root/repo/cpython/Doc/venv/bin/python3: No module named pip
/root/repo/cpython/Doc/venv/bin/python3: No module named pip
The venv has been created in the ./venv directory

$ python3.7 -m venv venv1
Error: Command '['/root/repo/cpython/Doc/venv1/bin/python3.7', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
```

So I was wondering we could further simplify the instruction by removing make venv and let the users create a virtual env with tools they are comfortable with, e.g. virtualenv, pipenv etc. 

And we could have streamlined steps for all platforms with something like this:
```
<Create a virtual environment>
cd Doc
python -m pip install --upgrade -r requirements.txt
# build using make
make html
# build using sphinx directly
python -m sphinx -b html . build/html
```

It is just my personal experience for your reference. Thanks

Reply:
Hmm, thanks for the feedback. Personally, I mostly use Conda environments for development, including to build the CPython docs, devguide, PEPs, etc, and thus don't really use the Makefile at all. myself. And it would certainly be nice to simplify this down to one cross-platform workflow...but unfortunately, there are a lot of complexities here that make that not really possible.

Switching to telling the user to create an arbitrary environment using their preferred environment manager requires either:

* Major modifications to the implementation and usage of the Makefile that would break existing workflows, since it assumes that a venv-created virtual environment is present at a specific name and location with the required dependencies, or
* Discarding the whole Makefile-based approach completely, and just recommending the `python -m sphinx` approach.

*Perhaps* one of these two options could be considered, but would first require a whole separate discussion from just improving the devguide documentation on the existing approaches.

Furthermore, I'm really not sure either of those approaches would be a good idea, as they would substantially complicate environment setup for the bulk of Make users, especially for doc contributors who may or may not be experienced in Python package management, as it requires them to know/learn how to create an environment and activate it on their particular platform, and furthermore ensure it is always activated before building the docs, as opposed to running a single command once.

In fact, the `venv` target could be modified to work like the one we have in the PEPs repo, which is automatically run as a dependency of the default build targets, automatically creating the environment if it doesn't exist and existing cleanly otherwise, such that users don't need to do any setup step at all—just run `make html` and It Just Works™. But, that's a subject for another issue...

This likely benefits many more users than those affected by Debian's persistent and self-inflicted Python packaging problems, or those wanting to build the docs with their own custom Python. For those users who've installed their own custom Python and want to run it with that, building directly with `python -m sphinx` inside an environment created with that Python is really the way to go.

> And we could have streamlined steps for all platforms with something like this:

Unfortunately, beyond the above, this is missing a key step—_activating_ the environment, which (with user-created environments) must be performed every time rather than just once, and like creating it, will vary depending on the type of environment. It's also missing a lot of key details from the original, like clarifying that `make` is available on both Windows and *nix and the difference between the two, what other commands can be used, how to get help, when `make` should be used vs. direct invocation, making unambiguous which steps are run once vs. every session, and which are specific to make or not, etc. Also, putting all the commands in one block without explaining the purpose of each step doesn't help users understand what they are doing and why, particularly important when they run into trouble or want to do something beyond the specific task stated. Additionally, it's missing the links and cross-references from the original that the user can consult for more information and guidance.

Reply:
@CAM-Gerlach Thanks! Let's leave it as is.